### PR TITLE
dbeaver/pro#2256 [Mac] Flipped splash screen on mac #2256

### PR DESCRIPTION
**Pull request by serjiokov**
OS Sonoma.

### CODE REVIEW

**Comment by serjiokov** [comment]
Jenkins build 
https://build.dev.dbeaver.com/job/DBeaverDesktop/job/MacBuilds/job/dbeaver-mac-builds/1072/

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java [hunk 1]
```diff
@@ -251,7 +252,10 @@ public Object start(IApplicationContext context) {
         writeWorkspaceInfo();
 
         // Update splash. Do it AFTER platform startup because platform may initiate some splash shell interactions
-        updateSplashHandler();
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java:254]: Please avoid copy-paste. Create function isMacOsSanoma.

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java:254]: Method extracted.

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java [hunk 2]
```diff
@@ -62,7 +63,13 @@ public DBeaverSplashHandler()
     @Override
     public void init(Shell splash) {
         super.init(splash);
-
+        /*
+         * Related to a bug in MacOS Sonoma
+         * (https://github.com/eclipse-platform/eclipse.platform.swt/issues/772)
+         */
+        if (isMacOsSomona()) { //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java]:
```suggestion
        if (isMacOsSomona()) {
```

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java [hunk 3]
```diff
@@ -72,6 +79,22 @@ public void init(Shell splash) {
...
 
     }
+    
+    @Override
+    public IProgressMonitor getBundleProgressMonitor() {
+        /*
+         * Related to a bug in MacOS Sonoma
+         * (https://github.com/eclipse-platform/eclipse.platform.swt/issues/772)
+         */
+        if (isMacOsSomona()) { //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java]:
```suggestion
        if (isMacOsSomona()) {
```

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverSplashHandler.java]: +

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java (hunk 1 above)
Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java:254]: It is still here

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/DBeaverApplication.java:254]: Moved to RuntimeUtils

**Comment by Matvey16** [comment]
Verified

**Pull request merged**